host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1034,42 +1036,105 @@ static FILE *open_filename(RASPIVID_STATE *pState, char *filename)
 
    if (filename)
    {
-      int network = 0, socktype;
+      int sfd, network = 0, network_listen = 0, socktype;
+
       if(!strncmp("tcp://", filename, 6))
       {
          network = 1;
          socktype = SOCK_STREAM;
       }
-      if(!strncmp("udp://", filename, 6))
+      else if(!strncmp("udp://", filename, 6))
       {
          network = 1;
          socktype = SOCK_DGRAM;
       }
-      if(network)
+      else
       {
-         filename += 6;
-         char *colon;
-         colon = strchr(filename, ':');
-         int sfd = socket(AF_INET, socktype, 0);
-         if(sfd < 0)
+         const char ltcpSrch[] = "ltcp://";
+         if (!strncmp(ltcpSrch, filename, sizeof(ltcpSrch) - 1))
          {
-              perror("socket");
-         }
+            network = network_listen = 1;
+            struct sockaddr_in serv_addr, cli_addr;
+            bzero((char *) &serv_addr, sizeof(serv_addr));
+
+            unsigned char* chp = (unsigned char*) &serv_addr.sin_addr.s_addr;
+            if (5 != sscanf(filename + sizeof(ltcpSrch) - 1,
+                            "%"SCNu8".%"SCNu8".%"SCNu8".%"SCNu8":%"SCNu16, chp + 0, chp + 1, chp + 2,
```

Comment:
We already do the equivalent for normal network connections using inet_aton at 1123. That's the normal call used for converting host names to addresses. Any reason we need to roll our own here?